runtime/src/snapshot_utils.rs

```diff
@@ -1672,7 +1668,48 @@ pub fn snapshot_bank(
     )
     .expect("failed to hard link bank snapshot into a tmpdir");
 
-    accounts_package_sender.send(accounts_package)?;
+    let can_submit = match snapshot_type {
+        // if the new AccountsPackage is for a full snapshot, always submit
+        Some(SnapshotType::FullSnapshot) => true,
+        // if the new AccountsPackage is for an incremental snapshot, submit as long as there isn't
+        // a pending full snapshot
+        Some(SnapshotType::IncrementalSnapshot(_)) => pending_accounts_package
+            .lock()
+            .unwrap()
+            .as_ref()
+            .and_then(|old_accounts_package| {
+                old_accounts_package
+                    .snapshot_type
+                    .map(|old_snapshot_type| !old_snapshot_type.is_full_snapshot())
+            })
+            .unwrap_or(true),
+        // otherwise, only submit the new AccountsPackage as long as there's not a pending package
+        // destined for a snapshot archive
+        None => pending_accounts_package
+            .lock()
+            .unwrap()
+            .as_ref()
+            .map(|old_accounts_package| old_accounts_package.snapshot_type.is_none())
+            .unwrap_or(true),
+    };
+
+    if can_submit {
+        let old_accounts_package = pending_accounts_package
+            .lock()
+            .unwrap()
+            .replace(accounts_package);
+        if let Some(old_accounts_package) = old_accounts_package {
+            debug!(
+                "The pending AccountsPackage has been overwritten: \
+                \nNew AccountsPackage slot: {}, snapshot type: {:?} \
+                \nOld AccountsPackage slot: {}, snapshot type: {:?}",
+                root_bank.slot(),
+                snapshot_type,
+                old_accounts_package.slot,
+                old_accounts_package.snapshot_type,
+            );
+        }
+    }
```

Comment:
a unit test for this case would be good to uphold the invariant

Reply:
Done!